### DISCUSSION TITLE
Prevent rayon from causing OOM-Sigkill

### DISCUSSION THREAD

**Discussion by mwurps** [rayon-rs/rayon#1177] [Q&A]
Hi all,

I was trying out rayon to speed up a computation:

`
let result = seeds.par_iter().map(|x| seed_to_location(&maps, *x)).min();
`

I want to transform all elements of a pretty large vector (*seeds*) with a function (*seeds_to_location*) and find the minimum of the transformed values. I thought this would be a good use case to try out rayon. However, I found that this computation consumes all of my RAM, until it gets inevitably shot down with a SIGKILL signal.

Is there a way to avoid this from happening with rayon or to lower it's memory consumption?

Thanks :)

**Comment by cuviper** [comment]
Rayon doesn't allocate much memory itself. Does `seed_to_location` use much memory? What is the peak memory usage if you run that as a sequential `.iter()`? Or with fewer threads like `env RAYON_NUM_THREADS=2`?
